app/scripts/controllers/util/error.js

```diff
@@ -40,4 +42,29 @@ angular.module('openshiftConsole')
     $scope.reloadConsole = function() {
       $window.location.href = "/";
     };
+
+    function checkHomePagePreference(projectName) {
+      if (projectName && projectName === HomePagePreference.getProjectOverviewPage()) {
+        HomePagePreference.clear();
+        NotificationsService.addNotification({
+          id: "invalid-home-page-preference",
+          type: "warning",
+          message: "Home page set to invalid project.",
+          details: "Project '" + projectName + "' no longer exists.",
+          links: [{
+            href: "",
+            label: "Set Home Page",
+            onClick: function() {
+              $uibModal.open({
```

Comment:
Unfortunately this is going to leak memory because of https://github.com/openshift/origin-web-common/issues/97. I'm not sure what to suggest though until that issue is fixed.

Reply:
> Unfortunately this is going to leak memory because of openshift/origin-web-common#97. I'm not sure what to suggest though until that issue is fixed.

Actually if you move the code that shows the notification into a service, I think it should be OK.